sw/device/lib/testing/sysrst_ctrl_testutils.h

```diff
@@ -12,17 +12,41 @@
 
 /**
  * Setup the sysrst_ctrl direct IO (DIO) pins, that is flash write protect (WP)
- * and EC reset. Also disable the EC reset debounce timer and release the flash
- * WP so that the output is not overidden to low by the block.
+ * and EC reset.
  *
  * This function will try to setup the DIO pins as open drain. If this fails,
  * it will then try virtual open drain.
  *
  * This function will panic on error.
  *
  * @param pinmux Pointer to an initialized pinmux DIF.
+ */
+void sysrst_ctrl_testutils_setup_dio(dif_pinmux_t *pinmux);
+
+/**
+ * Release the flash WP and EC reset so that the output is not overidden to low
+ * by the block.
+ *
+ * This function will panic on error.
+ *
+ * @param pinmux Pointer to an initialized sysrst_ctrl DIF.
+ * @param release_ec Whether to release the EC reset pin.
+ * @param release_flash Whether to release the flast write protect pin.
+ */
+void sysrst_ctrl_testutils_release_dio(dif_sysrst_ctrl_t *sysrst_ctrl,
+                                       bool release_ec, bool release_flash);
+
+/**
+ * Set the EC reset pulse width.
+ *
+ * This function will panic if the pulse width does not fit into the regiter.
```

Comment:
I guess there's a problem with units here. We need the pulse width, as measured in AON clock cycles, to fit in the 16 bit register.

Also nit: Missing "s" in "register".

Maybe something like this?

The supplied pulse width is converted into a count of AON clock ticks. If the width is not an exact multiple of the AON clock period, the conversion rounds down. The function will panic if the converted pulse width does not fit into the RST_CTL register.

Reply:
Thanks Rupert, that's indeed much clearer.